### ISSUE TITLE
gripper action causes very slow speeds

### ISSUE THREAD

**Issue by AlexanderKoch-Koch** [stepjam/RLBench#77]
The step function in task_environment.py is very slow if there is a gripper action (about 5x - 10x slower). Is there a reason for this while loop?
```
while not done:  
    done = self._robot.gripper.actuate(ee_action, velocity=0.2)  
    self._pyrep.step()  
    self._task.step()  
```
If I change it to execute the gripper.actuate function only once, it runs much faster.

And why is no arm action executed if there is a gripper action? Can't they be executed at the same time?

**Comment by AlexanderKoch-Koch** [comment]
I have seen that my last point has already been fixed in 1.0.7. I was still running 1.0.6.

However, I still don't understand why the gripper action has to be run in a loop.

**Comment by stepjam** [comment]
The gripper action has to be run in a loop because the gripper closes at a certain velocity; meaning that we have to step the simulation several times in order to close the gripper fully.
